Add readme and setup instructions

jscgh opened the pull request:
- Adds README
- Lets more variables be defined by the `.env` file

tlitfin-unsw commented on README.md:
```diff
@@ -0,0 +1,88 @@
+DEBUGGROUP=sbf-pipelines
+
+# Debug directory
+DEBUGDIR=/srv/scratch/sbf/debug/${RUNNAME}
```
I think RUNNAME won't be set when this is run? Probably need to set a BASE_DEBUGDIR with RUNNAME set at run time?

tlitfin-unsw commented on README.md:
```diff
@@ -0,0 +1,88 @@
+DEBUGDIR=/srv/scratch/sbf/debug/${RUNNAME}
+
+# Nextflow work directory
+NXF_WORK=/srv/scratch/${USER}/.proteinfold/work/${RUNNAME}
```
Same here regarding RUNNAME

tlitfin-unsw commented on README.md:
```diff
@@ -0,0 +1,88 @@
+# Example .env for ood-proteinfold
+
+# Output directory
+OUT_DIR=/srv/scratch/${USER}/proteinfold_output/<run_dir>
```
Probably only want to set a BASE_OUT_DIR and then the job-specific OUT_DIR is set relative to BASE_OUT_DIR?

tlitfin-unsw commented on README.md:
```diff
@@ -0,0 +1,88 @@
+# Nextflow work directory
+NXF_WORK=/srv/scratch/${USER}/.proteinfold/work/${RUNNAME}
+
+# Apptainer/Singularity blob cache directories (optional if set in your own config)
+# APPTAINER_CACHEDIR=/srv/scratch/${USER}/.images
+# SINGULARITY_CACHEDIR=/srv/scratch/${USER}/.images
+
+# Nextflow Apptainer/Singularity image cache/library directories (optional if set in your own config)
```
Could we make it clearer what we mean by "own config". Do we mean the institutional nextflow configs? Maybe we could link to the unsw_katana nextflow config?

tlitfin-unsw reviewed (changes requested):
Looking pretty nice.

I think we might need to also document the environment setup for samplesheet utils?

Could we also add a one line comment to provide context on each of the environment variables set in the .env file? eg project root is the location of local nextflow configs and the samplesheet-utils environment.

jscgh commented:
Addressed comments:

https://github.com/Australian-Structural-Biology-Computing/ood-proteinfold/pull/57/files/f369e0d30d0c1efdd53e50082cd9c7a7e292de9c..dad8473e585347e58313b51c52dd231761f792cd

Test runs completed successfully for AF2, ESMFold, and RFAA.

tlitfin-unsw commented on template/after.sh.erb:
```diff
@@ -1,5 +1,6 @@
+source .env
 RUN_DIR=$(echo "<%= context.run_name %>" | sed "s#[^A-Za-z0-9]#_#g")
-OUT_DIR="/srv/scratch/${USER}/proteinfold_output/${RUN_DIR}"
+OUT_DIR="${OUT_DIR:-/srv/scratch/${USER}/proteinfold_output/${RUN_DIR}}"
```
I think OUT_DIR no longer exists as an environment variable to be set in env. Probably need to change this logic to BASE_OUT_DIR.

jscgh replied:
Good catch, changed that out and doing a test run.

jscgh replied:
Passed the test run!

tlitfin-unsw commented on template/after.sh.erb:
```diff
@@ -1,5 +1,6 @@
+source .env
```
Do we want a check if .env exists? Could throw a warning if .env doesnt exist rather than:
`bash: .env: No such file or directory`

I notice we dont do any special handling in script.sh.erb so happy to leave it as is.
